sklearn/compose/_column_transformer.py

```diff
@@ -509,6 +522,14 @@ def transform(self, X):
         """
         check_is_fitted(self, 'transformers_')
 
+        # No column reordering allowed for named cols combined with remainder
+        if self._remainder[2] is not None and hasattr(self, '_df_columns'):
+            if len(X.columns) != len(self._df_columns) \
+               or any(X.columns != self._df_columns):
```

Comment:
Should we also allow for appended columns to maintain backwards compatibility?

Reply:
No problem with that from my side. The current check is stricter than really necessary for this fix anyway. I will add a commit to change it and reflect it in the unit tests.
BTW: who commonly clicks  "resolve conversation", the reviewer or the PR author?

Reply:
I don't regularly use the "resolve conversation" feature